agent/installer/installer.iss.in

```diff
@@ -0,0 +1,1597 @@
+// Validates required fields, formats, and values for host, endpoint, logging, encryption, and plugin source.
+// Logs errors to standard output and returns False if any check fails.
+// Returns True if all parameters are valid.
+function CheckSilentParameterValidity: Boolean;
+var
+  HostParam, EndpointParam, LogTypeParam, LogFileParam,Loglvl: string;
+  Pluginsrc_p: string;
+  EncryptionEnabled,PollerInitiatedEnabled: Boolean;
+begin
+  if not IsSilentMode then
+  begin
+    Result := True;
+    Exit;
+  end;
+
+  HostParam := ExpandConstant('{param:host|}');
+  EndpointParam := ExpandConstant('{param:endpoint|}');
+  LogTypeParam := LowerCase(ExpandConstant('{param:logtype|event-log}'));
+  LogFileParam := ExpandConstant('{param:logfile|}');
+  EncryptionEnabled := (LowerCase(ExpandConstant('{param:encryption|no}')) = 'full') or
+                       (LowerCase(ExpandConstant('{param:encryption|no}')) = 'insecure');
+  PollerInitiatedEnabled := (LowerCase(ExpandConstant('{param:reverse|false}')) = 'true');
+
+  // HOST
+  if HostParam = '' then
+  begin
+    LogToStdOutFile('Missing parameter: /HOST');
+    Result := False;
+    Exit;
+  end;
+
+  // ENDPOINT format host:port
+  if (Pos(':', EndpointParam) = 0) or (Copy(EndpointParam, Length(EndpointParam), 1) = ':') then
+  begin
+    LogToStdOutFile('Invalid format for /ENDPOINT. Expected "host:port".');
+    Result := False;
+    Exit;
+  end;
+
+  // if logtype = file → log file mandatory
+  if LogTypeParam = 'file' then
+  begin
+    if (LogFileParam = '') or not(IsPathValid(LogFileParam)) then
+    begin
+      LogToStdOutFile('Missing /LOGFILE when /LOGTYPE=file');
+      Result := False;
+      Exit;
+    end;
+  end;
+
+  // check log level 
+  Loglvl := LowerCase(ExpandConstant('{param:loglevel|error}'));
+  if (Loglvl <> 'off') and (Loglvl <> 'critical') and (Loglvl <> 'error') and (Loglvl <> 'warning') and (Loglvl <> 'info') and (Loglvl <> 'debug') and (Loglvl <> 'trace') then
+  begin
+    LogToStdOutFile('Invalid /LOGLEVEL value. Allowed values: off|critical|error|warning|info|debug|trace');
+    Result := False;
+    Exit;
+  end;
+
+  // if encryption test cert key token
+  if EncryptionEnabled then
+  begin
+    if PollerInitiatedEnabled then begin
+      if ((ExpandConstant('{param:cert|}') <> '') and not(IsPathValid(ExpandConstant('{param:cert|}')))) or (ExpandConstant('{param:cert|}') = '') then
+      begin
+        LogToStdOutFile('Missing /CERT when /ENCRYPTION=full/insecure');
+        Result := False;
+        Exit;
+      end;
+
+      if ((ExpandConstant('{param:key|}') <> '') and not(IsPathValid(ExpandConstant('{param:key|}')))) or (ExpandConstant('{param:key|}') = '') then
+      begin
+        LogToStdOutFile('Missing /KEY when /ENCRYPTION=full/insecure');
+        Result := False;
+        Exit;
+      end;
+     end
+    else begin
+      if not(ExpandConstant('{param:ca|}') <> '') and not(IsPathValid(ExpandConstant('{param:ca|}'))) then
```

Comment:
Perhaps I does not understand well, but it should not be if ExpandConstant('{param:ca|}') <> '' and not(IsPathValid(ExpandConstant('{param:ca|}'))) then